v2/features/farms/FarmTableRowBodyTransactionControls.tsx

```diff
@@ -26,10 +32,16 @@ const FarmsTableRowBodyTransactionControls: FC<Props> = ({ asset }) => {
   const account = useAccount();
 
   const isUserModelLoading = useAppSelector(UserSelectors.selectIsFetching);
+
+  // TODO: Relying on this only shows zap options for component tokens user holds in wallet - could be improved by persisted options (even with 0 balance)
```

Comment:
@bcstryker as described, we need to present the zap option based on a global object, and not simply the user's `UserModel` because this could be empty in a lot of cases